extensions/amp-wistia-player/0.1/amp-wistia-player.js

```diff
@@ -88,25 +119,88 @@ class AmpWistiaPlayer extends AMP.BaseElement {
       this.iframe_ = null;
     }
 
+    if (this.unlistenMessage_) {
+      this.unlistenMessage_();
+    }
+
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+
     return true;
   }
 
+  /** @override */
+  viewportCallback(visible) {
+    this.element.dispatchCustomEvent(VideoEvents.VISIBILITY, {visible});
+  }
+
   /** @override */
   pauseCallback() {
     if (this.iframe_) {
       this.pause();
     }
   }
 
+  /** @private */
+  handleWistiaMessages_(event) {
+    if (event.origin !== 'https://fast.wistia.net' ||
```

Comment:
Do we need to do the origin check? Would be hard to test this with local embeds, and sometimes they're hosted on fast.wistia.com or fast.wistia.st too.

Reply:
My take is that this is easily changeable when amphtml is in local dev since we won't be able to technically insert a different origin from what is on google's amp servers (prod).

Reply:
Point taken considering the maintenance perspective. 👍